### PR TITLE
Modify test_check_mon_pdb_post_upgrade with the new mon pdb changes

### PR DESCRIPTION
**Pull request by Akarsha-rai**
Fixes: #4178

Signed-off-by: Akarsha <akrai@redhat.com>

### CODE REVIEW

File: tests/ecosystem/upgrade/test_upgrade.py [hunk 1]
```diff
@@ -81,10 +82,15 @@ def test_check_mon_pdb_post_upgrade():
...
-    disruptions_allowed, min_available_mon = get_mon_pdb()
+    disruptions_allowed, min_available_mon, max_unavailable_mon = get_mon_pdb()
     log.info(f"Number of Mons Disruptions_allowed {disruptions_allowed}")
     log.info(f"Minimum_available mon count {min_available_mon}")
+    log.info(f"Minimum_available mon count {max_unavailable_mon}")
 
     # The PDB values are considered from OCS 4.5 onwards.
     assert disruptions_allowed == 1, "Mon Disruptions_allowed count not matching"
-    assert min_available_mon == 2, "Minimum available mon count is not matching"
+    if get_ocs_version() <= "4.6":
```
Comment (AviadP) [inline tests/ecosystem/upgrade/test_upgrade.py:92]:
```suggestion
    if get_ocs_parsed_version() <= 4.6:
```

**Comment by AviadP** [comment]
as general note, def test_check_mon_pdb_post_upgrade(): should not be placed in test_upgrade file, as it is not running any upgrade process. tests that should run post_upgrade, need to be placed in their path/file based on their testing topic. (in tests/e2e/ or tests/manage)

File: tests/ecosystem/upgrade/test_upgrade.py (hunk 1 above)
Comment (Akarsha-rai) [reply tests/ecosystem/upgrade/test_upgrade.py]: done

File: tests/ecosystem/upgrade/test_upgrade.py [hunk 2]
```diff
@@ -81,10 +82,15 @@ def test_check_mon_pdb_post_upgrade():
...
-    disruptions_allowed, min_available_mon = get_mon_pdb()
+    disruptions_allowed, min_available_mon, max_unavailable_mon = get_mon_pdb()
     log.info(f"Number of Mons Disruptions_allowed {disruptions_allowed}")
     log.info(f"Minimum_available mon count {min_available_mon}")
+    log.info(f"Minimum_available mon count {max_unavailable_mon}")
 
     # The PDB values are considered from OCS 4.5 onwards.
     assert disruptions_allowed == 1, "Mon Disruptions_allowed count not matching"
-    assert min_available_mon == 2, "Minimum available mon count is not matching"
+    if get_ocs_parsed_version() <= 4.6:
```
Comment (petr-balogh) [inline tests/ecosystem/upgrade/test_upgrade.py:92]: This is not good solution. See the issue here:
https://github.com/red-hat-storage/ocs-ci/issues/4066

I didn't have time to start working on the issue in general. Once I do I will probably cover the function get_ocs_parsed_version to return semantic version. But for now please handle as suggested in the issue.

Comment (Akarsha-rai) [reply tests/ecosystem/upgrade/test_upgrade.py]: using semantic version now

**Review by PrasadDesala — DISMISSED** [review]

**Review by yosibsh — DISMISSED** [review]

**Review by Shrivaibavi — DISMISSED** [review]
LGTM

**Review by petr-balogh — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**